main.go

```diff
@@ -68,6 +74,68 @@ func main() {
 		os.Setenv("OS_CLIENT_CONFIG_FILE", *osClientConfig)
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	errChan := make(chan error, 1)
+
+	// Start the backend service.
+	if *cacheEnable {
+		go cacheBackgroundService(ctx, services, errChan, logger)
+	}
+
+	// Start the HTTP server.
+	go startHTTPServer(ctx, services, toolkitFlags, errChan, logger)
+
+	// Wait for an error from any service or a termination signal.
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+	select {
+	case err := <-errChan:
+		level.Error(logger).Log("err", "Shutting down due to error", "err", err)
+		cancel()
+	case <-sigChan:
+		level.Info(logger).Log("msg", "Termination signal received. Shutting down...")
+		cancel()
+	}
+
+}
```

Comment:
There's a bug in here:

```
golang-openstack-exporter.openstack-exporter --cache --cache-ttl 2s
ts=2024-03-28T04:31:02.457Z caller=main.go:67 level=error msg="openstack-exporter: error: required argument 'cloud' or flag --multi-cloud not provided, try --help"
ts=2024-03-28T04:31:02.457Z caller=main.go:70 level=info msg="Build context" build_context="(go=go1.22.1, platform=linux/amd64, user=, date=, tags=unknown)"
ts=2024-03-28T04:31:02.457Z caller=main.go:106 level=info msg="Start cache background service"
ts=2024-03-28T04:31:02.457Z caller=utils.go:16 level=info msg="Run collect cache job"
ts=2024-03-28T04:31:02.457Z caller=main.go:152 level=info msg="openstack exporter started in legacy mode"
ts=2024-03-28T04:31:02.459Z caller=tls_config.go:274 level=info msg="Listening on" address=[::]:9180
ts=2024-03-28T04:31:02.459Z caller=tls_config.go:277 level=info msg="TLS is disabled." http2=false address=[::]:9180
ts=2024-03-28T04:31:03.458Z caller=utils.go:16 level=info msg="Run collect cache job"
ts=2024-03-28T04:31:04.457Z caller=utils.go:16 level=info msg="Run collect cache job"
ts=2024-03-28T04:31:04.457Z caller=main.go:129 level=info msg="Cache TTL flush"
ts=2024-03-28T04:31:05.457Z caller=utils.go:16 level=info msg="Run collect cache job"
ts=2024-03-28T04:31:06.458Z caller=utils.go:16 level=info msg="Run collect cache job"
ts=2024-03-28T04:31:06.458Z caller=main.go:129 level=info msg="Cache TTL flush"
^Cts=2024-03-28T04:31:06.564Z caller=main.go:99 level=info msg="Termination signal received. Shutting down..."
```
The background service did not stop if the exporter crashes if we don't provide <cloud> positional argument

Reply:
What is positional argument?

Reply:
This should fixable by some a early stage check.